### PR TITLE
Differential Fees

### PR DESCRIPTION
**Pull request by hav-noms**

### CODE REVIEW

File: SIPS/sip-56.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Currently there is one universal exchange rate for exchanging synths which is set onchain in the `FeePool`.
+
+Upgrading the `FeePool` to allow an array of Synths exchange rates to be set allowing unique pricing per asset or class.
+
+There could be multiple pricing options TBD.
```
Comment (justinjmoses) [inline SIPS/sip-56.md:27]: [np] "Such as:"

File: SIPS/sip-56.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+1. Charge the fee rate of the synth being exchanged into. e.g. If sETH is 1% and sUSD was 0% then it would be free to exchange from sETH to sUSD.
+2. Charge the sum of the pair. e.g. if sETH is .5% and sUSD is .1% the total exchange fee would be 0.6%
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+With a universal exchange fee rate the protocol is limited in it's ability to price markets fairly or apply incentive mechanisms. To deter front runners setting a high exchange rate was used to reduce profiitability affecting all Synth markets and discouraging good actors from trading.
```
Comment (justinjmoses) [inline SIPS/sip-56.md:36]: it's => its

File: SIPS/sip-56.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+Storage of the rates should use the `uint mappings` in the `FeePoolEternalStorage` contract so `FeePool` logic can be upgraded maintaining the exchange fee rate state.
+
+New functions required in the upgradable `FeePool` contract.
+
+### setExchangeFeeRateForSynths
```
Comment (justinjmoses) [inline SIPS/sip-56.md:54]: This is really good but what about how the fees are calculated?

Comment (hav-noms) [reply SIPS/sip-56.md:55]: Added the definition for exchangeFeeForTrade and to deprecate existing fee view function stub`FeePool.exchangeFeeIncurred`

**Pull request merged**